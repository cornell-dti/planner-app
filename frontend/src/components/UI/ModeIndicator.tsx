import React, { ReactElement } from 'react';
import systemMode, { Mode } from '../../util/mode-util';
import styles from './ModeIndicator.module.css';

type Props = { readonly mode: Mode };

/**
 * A mode indicator on the top-left corner to indicate whether a page is for DEV, STAGING or PROD.
 */
export const ConfigurableModeIndicator = ({ mode }: Props): ReactElement | null => {
  if (mode === 'PROD') {
    return null;
  }
  return <div className={styles.Indicator}>{mode}</div>;
};

export default (): ReactElement | null => <ConfigurableModeIndicator mode={systemMode} />;
