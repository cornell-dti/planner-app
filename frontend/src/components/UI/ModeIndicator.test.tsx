import React from 'react';
import renderer from 'react-test-renderer';
import ModeIndicator, { ConfigurableModeIndicator } from './ModeIndicator';

it('ConfigurableModeIndicator DEV matches snapshot.', () => {
  const tree = renderer.create(<ConfigurableModeIndicator mode="DEV" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ConfigurableModeIndicator STAGING matches snapshot.', () => {
  const tree = renderer.create(<ConfigurableModeIndicator mode="STAGING" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ConfigurableModeIndicator PROD matches snapshot.', () => {
  const tree = renderer.create(<ConfigurableModeIndicator mode="PROD" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ModeIndicator(DEV) matches snapshot.', () => {
  const tree = renderer.create(<ModeIndicator />).toJSON();
  expect(tree).toMatchSnapshot();
});
