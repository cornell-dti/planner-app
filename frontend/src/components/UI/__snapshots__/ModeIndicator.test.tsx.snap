// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigurableModeIndicator DEV matches snapshot. 1`] = `
<div
  className="Indicator"
>
  DEV
</div>
`;

exports[`ConfigurableModeIndicator PROD matches snapshot. 1`] = `null`;

exports[`ConfigurableModeIndicator STAGING matches snapshot. 1`] = `
<div
  className="Indicator"
>
  STAGING
</div>
`;

exports[`ModeIndicator(DEV) matches snapshot. 1`] = `
<div
  className="Indicator"
>
  DEV
</div>
`;
